### PR TITLE
add notification for dependent wearables being dropped

### PR DESCRIPTION
**Pull request by justinbrick**
## About the PR

added a notification for dependent wearables being dropped when the parent clothing is removed (i.e. PDA dropped when jumpsuit removed)

## Why / Balance

in relation to issue #28604, this gives a notification when the PDA is dropped. this also doubles purpose in giving a notification if any dependent wearables are dropped as well, 

## Technical details

this adds a single locale string & an additional check inside the unequip system of the inventory.

## Media

https://github.com/space-wizards/space-station-14/assets/49499807/242afe19-fdde-4e31-bc07-db4241939e66

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

None

**Changelog**

:cl:
- tweak: added a popup to notify when extra items are removed from your inventory while unequipping something

### CODE REVIEW

**Comment by justinbrick** [comment]
One thing to note is that the popup will likely still happen multiple times if there is multiple items dropped recursively.  a996847 tries to mitigate that, but doesn't guarantee it. i'm not sure if that can be fixed without changing the method signature (at the top of my head)

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 1]
```diff
@@ -424,15 +424,22 @@ public bool CanAccess(EntityUid actor, EntityUid target, EntityUid itemUid)
...
-                TryUnequip(actor, target, slotDef.Name, true, true, predicted, inventory, reparent: reparent);
+                if (TryUnequip(actor, target, slotDef.Name, true, true, predicted, inventory, reparent: reparent))
+                    droppedItems = true;
             }
         }
 
+        // we check if any items were dropped, and make a popup if they were.
+        // this doesn't guarantee that only 1 popup will be shown if multiple items were dropped in recursion.
+        if (droppedItems)
+            _popup.PopupPredicted(Loc.GetString("inventory-component-dropped-from-unequip"), target, target);
```
Comment (Tayrtahn) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:441]:
```suggestion
            _popup.PopupClient(Loc.GetString("inventory-component-dropped-from-unequip"), target, target);
```
PopupPredicted will show to everyone, PopupClient will show only to the specific player.

**Review by Tayrtahn — CHANGES_REQUESTED** [review]

**Comment by lzk228** [comment]
attached video doesnt load for me

**Comment by justinbrick** [comment]
> attached video doesnt load for me

I had a feeling that might be the case for some people. It's also not playing on discord for me, but it does play here. 
Give this one a shot, see if it works. If not, I might have to re-install OBS or try looking at my settings.

https://github.com/space-wizards/space-station-14/assets/49499807/0dda4b4e-0a8c-49cd-8d8d-35569cb1d064

**Comment by lzk228** [comment]
is it possible to actually display the dropped item(s)?

**Comment by deltanedas** [comment]
(the first video is hevc and some browsers cant play it for some reason)

**Comment by justinbrick** [comment]
> is it possible to actually display the dropped item(s)?

It is possible, but I don't think it's a very good idea, since you can drop multiple items at once - might cause a massive wall of text.

**Comment by lzk228** [comment]
> > is it possible to actually display the dropped item(s)?
> 
> It is possible, but I don't think it's a very good idea, since you can drop multiple items at once - might cause a massive wall of text.

then would be good to just add plural version `You dropped items!`

**Comment by Tayrtahn** [comment]
> then would be good to just add plural version `You dropped items!`

Implementation note: to do this properly, just pass the number of items as an argument when using Loc.GetString and handle picking the right text on the Fluent side.

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 2]
```diff
@@ -424,17 +443,27 @@ public bool CanAccess(EntityUid actor, EntityUid target, EntityUid itemUid)
...
             }
         }
 
-        if (!_containerSystem.Remove(removedItem.Value, slotContainer, force: force, reparent: reparent))
-            return false;
+        // we check if any items were dropped, and make a popup if they were.
+        // the reason we check for > 1 is because the first item is always the one we are trying to unequip,
+        // whereas we only want to notify for extra dropped items.
+        if (firstRun && itemsDropped > 1)
+            _popup.PopupClient(Loc.GetString("inventory-component-dropped-from-unequip", ("items", itemsDropped)), target, target);
```
Comment (Tayrtahn) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:466]: I recommend using `itemsDropped - 1` to make the localization more straightforward.

**Comment by SlamBamActionman** [comment]
Closed due to no response from PR author; has been continued in #33078.

**Pull request closed without merging**